torch/csrc/jit/codegen/cuda/lower_expr_sort.cpp

```diff
@@ -1,372 +1,963 @@
-  if (out->getValType() != ValType::TensorView) {
-    return nullptr;
+
+  ExprGroup(const ExprGroup& other)
+      : payload_(new ExprSortPayload(*(other.payload_))) {}
+
+  ExprGroup& operator=(const ExprGroup& other) {
+    *payload_ = *other.payload_;
+    exprs_ = other.exprs_;
+    return *this;
+  }
+
+  // Clears the traversal information in the payload
+  void clearTraversalInfo();
+
+  // Returns all neighbors, producers and consumers
+  std::vector<ExprGroup*> getNeighbors();
+
+  // Look at all neighbors of this and return who this could merge with based on
+  // level values of this, neighbors, and merged neighbors of neighbors
+  std::vector<ExprGroup*> getMergeCandidates();
+
+  std::unique_ptr<ExprSortPayload>& payload() {
+    return payload_;
+  }
+
+  const auto& producerEdges() const {
+    return producer_edges_;
+  }
+
+  void addProducerEdge(ExprGroupConnections* edge) {
+    addEdge(producer_edges_, edge);
+  }
+
+  void removeProducerEdge(ExprGroupConnections* edge) {
+    removeEdge(producer_edges_, edge);
+  }
+
+  void clearProducerEdges() {
+    producer_edges_.clear();
+  }
+
+  const auto& consumerEdges() const {
+    return consumer_edges_;
+  }
+
+  void addConsumerEdge(ExprGroupConnections* edge) {
+    addEdge(consumer_edges_, edge);
+  }
+
+  void removeConsumerEdge(ExprGroupConnections* edge) {
+    removeEdge(consumer_edges_, edge);
+  }
+
+  void clearConsumerEdges() {
+    consumer_edges_.clear();
+  }
+
+  auto& exprs() {
+    return exprs_;
+  }
+
+  const auto& exprs() const {
+    return exprs_;
+  }
+
+  const auto& inputVals() const {
+    return input_vals_;
+  }
+
+  void setInputVals(const std::vector<Val*>& input_vals) {
+    input_vals_ = input_vals;
+  }
+
+  void addInputVal(Val* input_val) {
+    input_vals_.push_back(input_val);
+  }
+
+  const auto& outputVals() const {
+    return output_vals_;
   }
-  return out->as<TensorView>();
-}
 
-struct TargetInfo {
-  TensorView* target = nullptr;
-  unsigned score = 0;
+  void setOutputVals(const std::vector<Val*>& output_vals) {
+    output_vals_ = output_vals;
+  }
+
+  void addOutputVal(Val* output_val) {
+    output_vals_.push_back(output_val);
+  }
+
+ private:
+  static void addEdge(
+      std::vector<ExprGroupConnections*>& edges,
+      ExprGroupConnections* edge_to_add) {
+    edges.push_back(edge_to_add);
+  }
+
+  static void removeEdge(
+      std::vector<ExprGroupConnections*>& edges,
+      ExprGroupConnections* edge_to_remove) {
+    auto it = std::find(edges.begin(), edges.end(), edge_to_remove);
+    TORCH_INTERNAL_ASSERT(it != edges.end(), "Could not find edge to remove.");
+    edges.erase(it);
+  }
+
+ private:
+  // "Ancestor nodes", towards inputs of segmentedDAG
+  std::vector<ExprGroupConnections*> producer_edges_;
+
+  // "Descendent nodes", towards outputs of segmentedDAG
+  std::vector<ExprGroupConnections*> consumer_edges_;
+
+  // TODO: Used?
+  std::vector<Val*> input_vals_;
+  std::vector<Val*> output_vals_;
```

Comment:
Are these used? They are set, but they do not seem to be used. Can they be removed?

Reply:
If they're not used, then yes, please remove.

Reply:
Removed.